sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -437,8 +437,7 @@ dmlStatementNoWith
         USING (source=multipartIdentifier |
           '(' sourceQuery=query')') sourceAlias=tableAlias
         ON mergeCondition=booleanExpression
-        matchedClause*
-        notMatchedClause*                                                          #mergeIntoTable
+        matchedOrNotMatchedClause*                                                 #mergeIntoTable
```

Comment:
This means people can mix the matched and notMatched clauses. Is it the same in other databases?

Reply:
I checked the SQL-Server and Oracle, both keep the WHEN MATCHED clauses before the WHEN NOT MATCHED clauses. Shall we obey the same constraint?

Reply:
yea I think so. Seems we can keep this parser rule unchanged.

Reply:
done.